CONTRIBUTING.md

```diff
@@ -31,24 +31,23 @@ There is a description of the [building and debugging process].
 
 If you want simply to have the `jenkins.war` file as fast as possible without tests, run:
 
-    mvn clean package -pl war -am -DskipTests -Dspotbugs.skip
+```sh
+mvn -am -pl war,bom -DskipTests -Dspotbugs.skip clean install
+```
 
 The WAR file will be created in `war/target/jenkins.war`.
 After that you can start Jenkins using Java CLI ([guide]).
 If you want to debug this WAR file without using Maven plugins,
 You can just start the executable with [Remote Debug Flags]
 and then attach IDE Debugger to it.
 
-To compile and launch a development instance, run:
+To launch a development instance, after the above command run:
 
-    mvn hudson-dev:run
+```sh
+mvn -f war jetty:run
```

Comment:
Is there a specific reason to use `-f` here?
This option exists to force Maven to use an alternative pom file instead of `pom.xml`. But here it is used to select the module. For this task, `-pl` is more appropriate.

Reply:
Does it matter? `-pl` probably also works, though I have not tried it.

Reply:
The one difference I know of: if after `install`ing everything as per prior instructions, you then make an edit to the root `pom.xml`, `-pl war` will reflect that edit while `-f war` will not. Not a typical situation, though.

Reply:
> Does it matter?

I think it does, for two reasons.

First, we produce unneccessary confusion. I was confused by the fact that one of the two steps for starting up Jenkins uses `-pl` and the other uses `-f`. I tried to figure out why this is important, so I had to consult the Maven help: 

    -f,--file <arg>       Force the use of an alternate POM file (or directory with pom.xml)
    -pl,--projects <arg>  Comma-delimited list of specified reactor projects to build instead
                          of all projects. A project can be specified by [groupId]:artifactId
                          or by its relative path

This did not reveal a compelling reason for using different flags for module selection.

Second (as you already mentioned above), `-pl` executes the module build in the context of a multimodule project and thus honours the current state of the parent pom. `-f` performs an isolated build of a module, using the installed (possibly outdated) version of the parent pom.
I do think that this difference will cause subtle problems to some less experienced users.